.pre-commit-config.yaml

```diff
@@ -9,10 +9,25 @@ repos:
         files: \.(TcPOU|TcDUT|TcGVL)$
 
 -   repo: https://github.com/pcdshub/pre-commit-hooks.git
-    rev: v1.3.0
+    rev: v1.7.0
     hooks:
     -   id: twincat-leading-tabs-remover
     -   id: twincat-lineids-remover
     -   id: twincat-xml-format
-    -   id: pytmc-pragma-linter
     -   id: check-fixed-library-versions
+    -   id: no-product-version
+    -   id: twincat-st-newline
+    # Check if minimize id changes is selected in the plc project file.
+    # See https://www.youtube.com/watch?v=KKpBtaYjfWo&t=935s why to do this.
+    # -   id: minimize-id-changes
+    # Checks if TwinCAT versions match in different tsproj files, or if it matches the targeted one.
+    -   id: check-twincat-versions
+        # Possible optional arguments
+        # --target-version: Set a version that you want the tsproj file to have
+        # --fix: Fix the version numbers if a target version is set
+        # --reason: Add a reason to the error message in case of a non-matching version.
+        # --pinned: Require the TwinCAT version to be pinned. Apply pinning if combined with --fix.
+        # --no-pinned: Require the TwinCAT version to not be pinned. Remove pinning if combined with --fix.
+        args: [--target-version=3.1.4024.20, --pinned, --fix, --reason="This version has a crucial new feature"]
```

Comment:
This version does not exist on the prog node
The reasoning for this should be changed to the actual reasoning

Reply:
#2 created to track.